### PR TITLE
Add pydantic-mongo

### PR DESCRIPTION
**Pull request by jonashaag**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jonashaag** [comment]
@conda-forge/help-python ready

File: recipes/pydantic-mongo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - pydantic >=2.0.2,<3.0.0
+
+test:
+  source_files:
+    - test/
+  imports:
+    - pydantic_mongo
+  commands:
+    - pip check
+    - pytest ./test -k 'not test_enhance_meta'  # [not win]
```
Comment (ocefpaf) [inline recipes/pydantic-mongo/meta.yaml:35]: Just curios but if this is noarch, what failed on Windows here?

Comment (jonashaag) [reply recipes/pydantic-mongo/meta.yaml:35]: See [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=964926&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4):

```
============================= test session starts =============================
platform win32 -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
rootdir: %SRC_DIR%
collected 0 items

============================ no tests ran in 0.03s ============================
ERROR: file or directory not found: test_enhance_meta'
```

I think there is something wrong with the `-k` parsing

Comment (ocefpaf) [reply recipes/pydantic-mongo/meta.yaml:35]: Ah! You may need to sue double quotes. Not a big deal for noarch packages but if it was an arch related failures, then we would need to do something about it.

**Pull request merged**